cni/pkg/nodeagent/net.go

```diff
@@ -50,15 +50,32 @@ var _ MeshDataplane = &NetServer{}
 // This cache will also be updated when the K8S informer gets a new pod.
 // This cache represents the "state of the world" of all enrolled pods on the node this agent
 // knows about, and will be sent to any connecting ztunnel as a startup message.
-func (s *NetServer) ConstructInitialSnapshot(ambientPods []*corev1.Pod) error {
+//
+// - For each of these existing snapshotted pods, steps into their netNS, and reconciles their
+// existing iptables rules against the expected set of rules. This is used to handle reconciling
+// iptables rule drift/changes between versions.
+func (s *NetServer) ConstructInitialSnapshot(existingAmbientPods []*corev1.Pod) error {
 	var consErr []error
 
-	podsByUID := slices.GroupUnique(ambientPods, (*corev1.Pod).GetUID)
+	podsByUID := slices.GroupUnique(existingAmbientPods, (*corev1.Pod).GetUID)
 	if err := s.buildZtunnelSnapshot(podsByUID); err != nil {
 		log.Warnf("failed to construct initial ztunnel snapshot: %v", err)
 		consErr = append(consErr, err)
 	}
 
+	if s.podIptables.ReconcileModeEnabled() {
+		log.Debug("inpod reconcile mode enabled")
```

Comment:
I think we will need to add a disclaimer to the [CNI upgrading doc](https://istio.io/latest/docs/setup/additional-setup/cni/) on istio.io about the potential traffic impact that could occur if the initial iptables rules differ from those in the target version.

Reply:
We will, yes - we have always documented that ambient dataplane upgrades in general may impact traffic, but should add more detail here. 


I plan to write/update the docs, either after this PR is merged, or before it is merged and after it is okayed.